vassal-app/src/main/java/VASSAL/build/module/GameState.java

```diff
@@ -567,17 +567,19 @@ public void setup(boolean gameStarting) {
 
     if (gameStarted) {
       if (gameStarting) {
+        // Ask the IndexManager to rebuild all indexes so at-start stack pieces are all included
+        // This is required for any SGKC's to work
+        GameModule.getGameModule().getIndexManager().rebuild();
```

Comment:
Possibly you should put this inside the invokeLater, just before the SGKCs? I can't remember the exact detail that leads us to invokeLater "to make sure SGKCs occur after all the other setup" - is something potentially loading in a thread?

Reply:
Probably safest. Thanks.